gstreamer-plugin/git-patch/0001-svthevcenc-Add-svthevc-encoder-element.patch

```diff
@@ -0,0 +1,2123 @@
++ *
++ **/
++
++#ifdef HAVE_CONFIG_H
++#  include "config.h"
++#endif
++
++#include "gstsvthevcenc.h"
++
++#include <gst/pbutils/pbutils.h>
++#include <gst/video/video.h>
++#include <gst/video/gstvideometa.h>
++#include <gst/video/gstvideopool.h>
++#include <gst/base/gstbitreader.h>
++
++#include <string.h>
++#include <stdlib.h>
++
++GST_DEBUG_CATEGORY_STATIC (svthevc_enc_debug);
++#define GST_CAT_DEFAULT svthevc_enc_debug
++
++enum
++{
++  PROP_0,
++  PROP_INSERT_VUI,
++  PROP_AUD,
++  PROP_HIERARCHICAL_LEVEL,
++  PROP_LOOKAHEAD_DISTANCE,
++  PROP_ENCODER_MODE,
++  PROP_RC_MODE,
++  PROP_QP,
++  PROP_QP_MAX,
++  PROP_QP_MIN,
++  PROP_SCENE_CHANGE_DETECTION,
++  PROP_TUNE,
++  PROP_LATENCY_MODE,
++  PROP_BASE_LAYER_SWITCH_MODE,
++  PROP_BITRATE,
++  PROP_KEY_INT_MAX,
++  PROP_INTRA_REFRESH,
++  PROP_CORES,
++  PROP_SOCKET,
++  PROP_TILE_ROW,
++  PROP_TILE_COL,
++};
++
++#define PROP_INSERT_VUI_DEFAULT             FALSE
++#define PROP_AUD_DEFAULT                    FALSE
++#define PROP_HIERARCHICAL_LEVEL_DEFAULT     3
++#define PROP_LOOKAHEAD_DISTANCE_DEFAULT     40
++#define PROP_ENCODER_MODE_DEFAULT           9
++#define PROP_RC_MODE_DEFAULT                0
++#define PROP_QP_DEFAULT                     25
++#define PROP_QP_MAX_DEFAULT                 48
++#define PROP_QP_MIN_DEFAULT                 10
++#define PROP_SCENE_CHANGE_DETECTION_DEFAULT TRUE
++#define PROP_TUNE_DEFAULT                   1
++#define PROP_LATENCY_MODE_DEFAULT           0
++#define PROP_BASE_LAYER_SWITCH_MODE_DEFAULT 0
++#define PROP_BITRATE_DEFAULT                (7 * 1000)
++#define PROP_KEY_INT_MAX_DEFAULT            -2
++#define PROP_INTRA_REFRESH_DEFAULT          1
++#define PROP_CORES_DEFAULT                  0
++#define PROP_SOCKET_DEFAULT                 -1
++#define PROP_TILE_ROW_DEFAULT               1
++#define PROP_TILE_COL_DEFAULT               1
++
++#define PROFILE_DEFAULT                     2
++#define LEVEL_DEFAULT                       0
++#define TIER_DEFAULT                        0
++
++#if G_BYTE_ORDER == G_LITTLE_ENDIAN
++#define FORMATS "I420, Y444, I420_10LE, Y444_10LE"
++#else
++#define FORMATS "I420, Y444, I420_10BE, Y444_10BE"
++#endif
++
++static GstStaticPadTemplate sink_factory = GST_STATIC_PAD_TEMPLATE ("sink",
++    GST_PAD_SINK,
++    GST_PAD_ALWAYS,
++    GST_STATIC_CAPS ("video/x-raw, "
++        "format = (string) { " FORMATS " }, "
++        "framerate = (fraction) [0, MAX], "
++        "width = (int) [ 64, 8192 ], " "height = (int) [ 64, 4320 ]")
++    );
++
++static GstStaticPadTemplate src_factory = GST_STATIC_PAD_TEMPLATE ("src",
++    GST_PAD_SRC,
++    GST_PAD_ALWAYS,
++    GST_STATIC_CAPS ("video/x-h265, "
++        "framerate = (fraction) [0/1, MAX], "
++        "width = (int) [ 64, 8192 ], " "height = (int) [ 64, 4320 ], "
++        "stream-format = (string) byte-stream, "
++        "alignment = (string) au, "
++        "profile = (string) { main, main-10, main-4:4:4 }")
```

Comment:
svt-hevc doesn't support 4:0:0 yet. So main-4:4:4 is not fully supported.

Reply:
But the svt encoder seems to force the MainREXT profile of output in EB_YUV444 format. 
```
SVT [Error]: Instance 1: The input profile is not correct, should be 4 or MainREXT for YUV422 or YUV444 cases
```
In this case, I think we need to specify main-4:4:4 in src caps to avoid negotiation fail.

Reply:
Sure, I will add it. Thx.